### PR TITLE
Smc fixes

### PR DESCRIPTION
**Pull request by whitewhim2718**
Made changes as discussed in Issue #41. Also added test case to test both the changes I made, and the SMC updater as a whole on the SimplePrecessionModel. Tests are stochastic, but for 100 runs with the test parameters, there were no fails.

### CODE REVIEW

**Comment by whitewhim2718** [comment]
It appears that my original PR did not have recent Travis config changes pulled from master, and passed Travis build on the fork, but not in the master. I've pulled in the Travis changes, and will make sure everything passes when I next get some time.

**Comment by scasagrande** [comment]
Just include `from __future__ import absolute_import` at the top of imports in `qinfer.tests.precession_model_test`

**Comment by cgranade** [comment]
Thank you for the submission, looks like a very nice addition! I have a few comments on the source itself, if you're willing to take care of them I'd appreciate, otherwise I'm happy to merge in as-is and I'll take care of them. Thank you again!

File: src/qinfer/smc.py [hunk 1]
```diff
@@ -1342,31 +1342,113 @@ def _bim(self, modelparams, expparams, modelweights=None):
...
+        Evaluates the local Bayesian Information Matrix (BIM) over all particles
+        of the current posterior distribution with corresponding weights. 
+
+        :param expparams: Parameters describing the experiment that was
+            performed.
+        :type expparams: :class:`~numpy.ndarray` of dtype given by the
+            :attr:`~qinfer.abstract_model.Model.expparams_dtype` property
+            of the underlying model
+
+        :param n_samples int: Number of samples to draw from particle distribution,
```
Comment (cgranade) [inline src/qinfer/smc.py]: The ``n_samples`` parameter only applies to the prior BIM method.

File: src/qinfer/smc.py [hunk 2]
```diff
@@ -1342,31 +1342,113 @@ def _bim(self, modelparams, expparams, modelweights=None):
...
         # If we're tracking the information content accessible to adaptive
         # algorithms, then we must use the current posterior as the prior
         # for the next step, then add that accordingly.
         if self._track_adaptive:
-            self.adaptive_bim += self.posterior_bayes_information(expparams)[:, :, 0]
+            self._adaptive_bim += self.posterior_bayes_information(expparams)[:, :, 0]
         
         # We now can update as normal.
-        SMCUpdater.update(self, outcome, expparams)
+        SMCUpdater.update(self, outcome, expparams,check_for_resample=True)
```
Comment (cgranade) [inline src/qinfer/smc.py]: This should be ``check_for_resample=check_for_resample``, to pass the argument through. Thanks for catching that this was not currently happening.

File: src/qinfer/tests/precession_model_test.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+		#Assert that models have learned true model parameters from data 
+		assert_almost_equal(self.updater.est_mean(),TestSMCUpdater.MODELPARAMS,2)
+		assert_almost_equal(self.updater_bayes.est_mean(),TestSMCUpdater.MODELPARAMS,2)
+		assert_almost_equal(self.num_updater.est_mean(),TestSMCUpdater.MODELPARAMS,2)
+		assert_almost_equal(self.num_updater_bayes.est_mean(),TestSMCUpdater.MODELPARAMS,2)
+
+
+		#test means
+		assert_almost_equal(self.updater.est_mean(),TestSMCUpdater.MODELPARAMS,2)
```
Comment (cgranade) [inline src/qinfer/tests/precession_model_test.py]: Are these lines redundant to the ones above?

File: src/qinfer/smc.py (hunk 2 above)
Comment (whitewhim2718) [inline src/qinfer/smc.py]: Careless of me to just put True. I'm not sure what I was thinking at the time.

File: src/qinfer/smc.py (hunk 1 above)
Comment (whitewhim2718) [inline src/qinfer/smc.py]: yes, I've fixed.

File: src/qinfer/tests/precession_model_test.py (hunk 3 above)
Comment (whitewhim2718) [inline src/qinfer/tests/precession_model_test.py]: yup.

**Comment by cgranade** [comment]
Thanks for addressing those few things, looks good! I'll go on and merge it in now, thanks again!

**Pull request merged**